NumLib/Function/Interpolation.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_INTERPOLATION_H
+#define NUMLIB_INTERPOLATION_H
+
+#include<array>
+#include<cassert>
+
+namespace NumLib
+{
+
+/**
+ * Interpolates variables given at element nodes according to the given shape matrix.
+ *
+ * This function simply does the usual finite-element interpolation, i.e. multiplication
+ * of nodal values with the shape function.
+ *
+ * @param nodal_values   vector of nodal values, ordered by component
+ * @param shape_matrix_N shape matrix of the point to which will be interpolated
+ * @param interpolated_values array of addresses to which the interpolated values will be written
+ */
+template<typename NodalValues, typename ShapeMatrix, std::size_t NodalDOF>
+void shapeFunctionInterpolate(
```

Comment:
The purpose of this function is to avoid writing the loop in each process. And I think, even with Eigen one would need a loop and, e.g. `block` expressions.